packages/connector/README.md

```diff
@@ -2,387 +2,119 @@
-
-```typescript
-saveCredentials(params: SaveCredentialsRequestParams): Promise<W3CCredential[]>
-```
-
-```typescript
-type SaveCredentialsRequestParams = {
-  body: {
-    credentials: [
-      {
-        description: string;
-        id: string;
-      },
-    ];
-    url: string;
-  };
-  from: string;
-  id: string;
-  thid?: string;
-  to: string;
-  typ?: string;
-  type: string;
-};
+export const zkpSnap = new ZkpSnap()
 ```
 
-Returns saved Verifiable Credentials
-
+### Establish connection
 ```typescript
-type W3CCredential = {
-  id: string;
-  '@context': string[];
-  type: string[];
-  expirationDate?: string;
-  issuanceDate?: string;
-  credentialSubject: { [key: string]: object | string | number };
-  credentialStatus: CredentialStatus;
-  issuer: string;
-  credentialSchema: CredentialSchema;
-  proof?: { [key: string]: any } | any[];
-};
-
-type CredentialStatus = {
-  id: string;
-  type: string;
-  revocationNonce?: number;
-  statusIssuer?: CredentialStatus;
-};
-
-type CredentialSchema = {
-  id: string;
-  type: string;
-};
+await zkpSnap.enable()
 ```
 
-### Create a proof
-
-Make sure you are on the correct network before creating a proof!
-To create a proof you need to call this method with params:
-
+### Check connection status
 ```typescript
-createProof(params: CreateProofRequestParams): Promise<ZKPProofResponse>
+const isInstalled = await zkpSnap.isInstalled()
 ```
 
+### Additional helpers
 ```typescript
-type CreateProofRequestParams = {
-  id?: number;
-  accountAddress?: string; // Metamask user address for on-chain proofs
-  circuitId:
-    | 'credentialAtomicQueryMTPV2'
-    | 'credentialAtomicQueryMTPV2OnChain'
-    | 'credentialAtomicQuerySigV2'
-    | 'credentialAtomicQuerySigV2OnChain';
-  challenge?: string; // bigint string
-  query: ProofQuery;
-};
+import { isMetamaskInstalled } from '@rarimo/rarime-connector'
 
-type ProofQuery = {
-  allowedIssuers?: string[];
-  credentialSubject?: { [key: string]: any };
-  schema?: string;
-  claimId?: string;
-  credentialSubjectId?: string;
-  context?: string;
-  type?: string;
-};
-```
+const init = async () => {
+  const isInstalled = await isMetamaskInstalled()
 
-Returns ZKPProofResponse - zkpProof for off-chain and updateStateTx, statesMerkleData, ZKProof for on-chain
-
-```typescript
-type ZKPProofResponse = {
-  updateStateTx?: TransactionRequest; // ethers TransactionRequest
-  zkpProof: ZKProof;
-  statesMerkleData?: {
-    issuerId: string;
-    state: StateInfo;
-    merkleProof: string[];
-  };
-};
-
-type ZKProof = {
-  proof: ProofData;
-  pub_signals: string[];
-};
-type ProofData = {
-  pi_a: string[];
-  pi_b: string[][];
-  pi_c: string[];
-  protocol: string;
-};
-type StateInfo = {
-  index: string;
-  hash: string;
-  createdAtTimestamp: string;
-  lastUpdateOperationIndex: string;
-};
+  if (isInstalled) {
+    // ...
+  }
+}
 ```
 
-### Check state contract sync
+## ZKP quick examples
```

Comment:
Split into separate snippets per function, like it was in the previous version. How to:
- add credential
- generate a ZK proof
- etc